### ISSUE TITLE
confused about implementation of 1-1, 1-n, n-1, n-n

### ISSUE THREAD

**Issue by junkangwu** [liuxiyang641/RAGAT#4]
hi, thanks a lot for your work in knowledge graph completion, but I still be confused about the implementation of the table5 in your paper.
1. as for the relation category, Following Wang et al. (2014), for each relation r, we compute the average number of tails per head (tphr) and the average number of head per tail (hptr). If tphr < 1.5 and hptr < 1.5, r is treated as one-to-one; if tphr ≥ 1.5 and hptr ≥ 1.5, r is treated as a many-to-many; if tphr < 1.5 and hptr ≥ 1.5, r is treated as one-to-many. So should we take the valid dataset and test dataset into consideration in this process? Or should we only classify them in the training dataset?
2. take tail prediction in 1-n relation category as for an example, should we choose the all 1-n relations prediction scores and take all the results into mean? 

I'm confused to re-implement this part of the experiment. It would be best if you could take a script as an example. Thanks a lot in advance!

**Comment by liuxiyang641** [comment]
1. In our experiments, we classify relations into four categories by analysising triples in the trainning data set. Then the test set can be spilts into 1-1, 1-m, m-1 and m-m subsets. 
2. Yes, the evaluating process for different relation categories just follows the same procedures that utilized in common link prediction task.  You can create different subsets and put them into the model.

**Comment by junkangwu** [comment]
Thanks a lot! I got it!
